src/components/UserSelector.tsx

```diff
@@ -1,16 +1,71 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  users: User[];
+  selectedUser: User | null;
+  setSelectedUser: (userToSelect: User) => void;
+  setSelectedPost: (postToSelect: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUser,
+  setSelectedUser,
+  setSelectedPost = () => {},
```

Comment:
Setting a default value for `setSelectedPost` to an empty function can lead to unexpected behavior if the prop is not provided. Consider making this prop required without a default value.